libs/image-editor/ImageEditor/src/main/kotlin/org/wordpress/android/imageeditor/preview/PreviewImageFragment.kt

```diff
@@ -36,22 +39,57 @@ class PreviewImageFragment : Fragment() {
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? = inflater.inflate(layout.fragment_preview_image, container, false)
+    ): View? = inflater.inflate(R.layout.preview_image_fragment, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         val nonNullIntent = checkNotNull(requireActivity().intent)
         initializeViewModels(nonNullIntent)
-        initializeViews(nonNullIntent)
+        initializeViews()
     }
 
-    private fun initializeViews(nonNullIntent: Intent) {
-        errorLayout.setOnClickListener {
-            val lowResImageUrl = nonNullIntent.getStringExtra(ARG_LOW_RES_IMAGE_URL)
-            val highResImageUrl = nonNullIntent.getStringExtra(ARG_HIGH_RES_IMAGE_URL)
+    private fun initializeViews() {
+        initializeViewPager()
+    }
 
-            viewModel.onLoadIntoImageViewRetry(lowResImageUrl, highResImageUrl)
+    private fun initializeViewPager() {
+        val previewImageAdapter = PreviewImageAdapter(
+            loadIntoImageViewWithResultListener = { imageData, imageView, position ->
+                loadIntoImageViewWithResultListener(imageData, imageView, position)
+            }
+        )
+        previewImageViewPager.adapter = previewImageAdapter
+
+        val tabConfigurationStrategy = TabLayoutMediator.TabConfigurationStrategy { tab, position ->
+            if (tab.customView == null) {
+                val customView = LayoutInflater.from(context).inflate(layout.preview_image_thumbnail, null)
+                tab.customView = customView
+            }
+            val imageView = (tab.customView as FrameLayout).findViewById<ImageView>(R.id.thumbnailImageView)
+            val pagerAdapter = previewImageViewPager.adapter as PreviewImageAdapter
+            val imageData = pagerAdapter.currentList[position].data
+            loadIntoImageView(imageData.lowResImageUrl, imageView)
```

Comment:
I'm thinking to pass selected position to the VM and let VM pass image data to the fragment to load the thumbnail. This selected position can also be used to manage selected position on orientation change.

Reply:
>This selected position inside VM can also be used to select the right thumbnail on orientation change.

Re-selection of the correct tab should be handled by the system. 

> let VM pass image data to the fragment to load the thumbnail

What do you mean by that? Thumbnail loading already works, right?

Reply:
I think it was premature to comment on it. Please ignore it. 

> Re-selection of the correct tab should be handled by the system.

Currently last selected tab doesn't get reselected after orientation change. It is in my pending action items list, might need some changes in the VM.